server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -102,17 +101,19 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
         }
     }
 
-    private Stream<MailboxId> retrieveMailboxIds(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
+    private Stream<Optional<Mailbox>> retrieveMailboxes(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
```

Comment:
> why do you want to change the return type ?

Because I needed to go one Stream operation further

> Do we care about retrieving absent mailboxes ?

No that's why the caller filter it.

Actually you might know that optinal unboxing is a pain with java8, it's not scala where you can simply do it with flatMap. Here, I let the caller (which is private) do the unboxing for my branching.